torch/_functorch/aot_autograd.py

```diff
@@ -684,9 +688,10 @@ def choose_dispatcher(needs_autograd, aot_config):
 
         compiler_fn = choose_dispatcher(needs_autograd, aot_config)
 
-        compiled_fn, fw_metadata = compiler_fn(
-            flat_fn, fake_flat_args, aot_config, fw_metadata=fw_metadata
-        )
+        with maybe_suppress():
+            compiled_fn, fw_metadata = compiler_fn(
+                flat_fn, fake_flat_args, aot_config, fw_metadata=fw_metadata
+            )
```

Comment:
This is especially unlikely to be correct, because Inductor can generate new guards and we really DO need to respect them, because they are necessary for correctness